### ISSUE TITLE
Bizzare error for invalid `if`

### ISSUE THREAD

**Issue by Y-Less** [pawn-lang/compiler#512]
### Issue description:

Missing the brackets on an `if` statement gives an error about invalid subscripts:

```
error 028: invalid subscript (not an array or too many subscripts): "var"
error 029: invalid expression, assumed zero
warning 215: expression has no effect
error 001: expected token: ";", but found "-end of file-"
```

This would be a far better error:

`error 001: expected token: "(", but found "-identifier-"`

### Minimal complete verifiable example (MCVE):

```pawn
main()
{
	new var;
	if var
	{
	}
}

```

### Workspace Information:

* Compiler version:
* Command line arguments provided (or sampctl version):
* Operating System:

3.10.10

**Comment by Daniel-Cortez** [comment]
Since there are no parentheses, the compiler expects keyword `*then` after the `if` control expression, e.g.:
```Pawn
if var *then *begin /* ... */ *end
```
I'm not sure how many people actually use this alternative syntax, but it's still valid in our version of Pawn (IIRC it was removed only in Pawn 3.3).

Anyway, since keyword `*then` is missing in the example as well, the compiler treats the opening `{` as a continuation of a control expression, as if `var` was subscripted as a byte array, hence the "invalid subscript" error.
In other words, the compiler works correctly.

**Comment by Daniel-Cortez** [comment]
@pawn-lang As I explained in the above post, the compiler works correctly, so this issue should probably be closed.
